security/middleware.py

```diff
@@ -655,42 +655,88 @@ def _csp_builder(self, csp_dict):
 
         return '; '.join(csp_components)
 
+    def _choose_csp_str(self, csp_dict, csp_str, err_msg):
+        """
+        Choose the Content-Security-Policy string to return.
+
+        Args:
+            csp_dict: a dictionary of values for building a CSP string
+            csp_str: the fallback CSP string if no dictionary is provided
+            err_msg: the message to log if both a dict and string are provided
+
+        Returns:
+        The Content-Security-Policy string by either building it from a
+        dictionary or using the provided string.
+        Log an error message if both are provided.
+        """
+        if csp_dict and csp_str:
+            logger.error('%s', err_msg)
+            raise django.core.exceptions.MiddlewareNotUsed
+
+        if csp_dict:
+            return self._csp_builder(csp_dict)
+        elif csp_str:
+            return csp_str
+        else:
+            return ''
+
     def __init__(self):
         # sanity checks
-        csp_mode = getattr(django.conf.settings, 'CSP_MODE', None)
+        self._csp_mode = getattr(django.conf.settings, 'CSP_MODE') or 'enforce'
```

Comment:
You still need the 'None'. If you don't supply CSP_MODE getattr will raise an exception.